packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1291,7 +1291,7 @@ export class DeltaManager
     // reuse.
     // Also payload goes to telemetry, so no PII, including content!!
     private comparableMessagePayload(m: ISequencedDocumentMessage) {
-        return `${m.clientId}-${m.type}-${m.minimumSequenceNumber}-${m.referenceSequenceNumber}`;
+        return `${m.clientId}-${m.type}-${m.minimumSequenceNumber}-${m.referenceSequenceNumber}-${m.timestamp}`;
```

Comment:
+@tanviraumi 

It's possible for a duplicate op to be broadcasted and have everything the same except the `timestamp`. That should be a supported scenario on the client.
I assume including the timestamp here would break that capability.

Reply:
Agree! Is there any specific reason to add "timestamp"? If so, we need to rethink how we add timestamp to the messages.

Reply:
Basically, we need to make sure that the service is not generating or updating the timestamp. Then it can always change on a restart.

Reply:
Yesterday at 10am meeting timestamp was called out as one of the unique pieces of info that is guaranteed to change (i.e. no way of repeating it if file is overwritten). Thus I've added it.
If service can broadcast same op but with different timestamp, then I'd remove it, but it seems a bit weird, in a sense that looking at file at different time produces slight differences, even if they are not substantially, the file (from client perspective) is changing in the past...

Reply:
I double checked how deli timestamps the ops. Originally op timestamps were not deterministic but I made [two](https://github.com/microsoft/FluidFramework/pull/2239) [prs](https://github.com/microsoft/FluidFramework/pull/2357) a while back to fix that. I forgot about those, my bad! So it should be fine to log the timestamp now.
Just be aware that the timestamp is not guaranteed to always move forward due to the possibility of frontend server clock skew, so don't validate based on that.

Reply:
Sounds good, we are not relying on it moving forward. @tanviraumi , any pushback to adding timestamp as comparison criteria back?

Reply:
Works for me!

@GaryWilber Are you using the same logic as r11s and adding timestamp in Kafka orderer. Or push does it before?

https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/kafka-orderer/src/kafkaOrderer.ts#L93

Reply:
@tanviraumi I'm using the same logic